Assets/GameAssets/Player/PlayerScripts/PlayerHealth.cs

```diff
@@ -43,8 +48,40 @@ public override void IncreaseHealth(float heal)
         base.IncreaseHealth(heal);
 
         PlayerManager.Instance.InvokePlayerHealthChangeEvent(GetHealthPercent(), _currentHealth);
+        CheckHeartBeat();
     }
 
+    private void CheckHeartBeat()
+    {
+        if (base._currentHealth <= _HeartBeatStart && base._currentHealth != _HeartBeatEnd && _heartBeatCoroutine == null)
+        {
+            StartCoroutine(HeartbeatLoop());
+        }
+        if (base._currentHealth == _HeartBeatEnd && _heartBeatCoroutine != null)
+        {
+            StopCoroutine(HeartbeatLoop());
+        }
+    }
+/// <summary>
+/// A coroutine that is mean't to loop the heartbeat sfx
+/// </summary>
+/// <returns>returns null to loop similar to update function</returns>
+    IEnumerator HeartbeatLoop()
+    {
+        float timer = 0.0f;
+        while (true)
+        {
+            if (timer > _HeartBeatRate)
+            {
+                RuntimeSfxManager.APlayOneShotSfx?.Invoke(FmodSfxEvents.Instance.PlayerHeartBeat,
+                    gameObject.transform.position);
+                timer = 0.0f;
+            }
+            timer += Time.deltaTime;
```

Comment:
Is this something we can just do with WaitForSeconds?

Reply:
^ that would be cleaner instead of making a timer. The `WaitForSeconds` can wait for` _heartBeatRate`